web/libs/datamanager/src/stores/AppStore.js

```diff
@@ -484,17 +484,18 @@ export const AppStore = types
       try {
         const newProject = yield self.apiCall("project", params);
         const projectLength = Object.entries(self.project ?? {}).length;
+        const newProjectLength = Object.entries(newProject ?? {}).length;
```

Comment:
The naming is really confusing here. better to rename it to something like `isProjectEmpty` and store boolean directly. We don't actually need a number of properties in project.